src/sql/azdata.proposed.d.ts

```diff
@@ -432,6 +432,23 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+	export interface PromptFailedResult {
+		/**
+		 * Signals that an error other than the user cancelled sign in error has occurred
+		 */
+		error?: boolean;
```

Comment:
Do you really need this? It seems like if cancelled is false then this should always be true so I'd just remove this and have cancelled be the determination here.

Reply:
In the happy path scenario an account is returned and cancelled is undefined I need the boolean

Reply:
Yeah I'd be returning an account, I agree actually that this is not needed so I'll remove it

Reply:
I think the problem is that this method returns the same thing for false/undefined
```
private isCanceledResult(result: azdata.Account | azdata.PromptFailedResult): result is azdata.PromptFailedResult {
		return (<azdata.PromptFailedResult>result).canceled;
	}
```

Reply:
You could change this to 

```
private isPromptFailedResult(result: azdata.Account | azdata.PromptFailedResult): result is azdata.PromptFailedResult {
		return typeof (<azdata.PromptFailedResult>result).canceled === 'boolean';
	}
```

And then update the check to first check if it's an errorResult and then have an inner if for whether canceled is true/false

Reply:
Still need to remove this property 😄